src/Reflection/ReflectionClass.php

```diff
@@ -171,9 +171,19 @@ public function __toString()
         return $str;
     }
 
-    public static function createFromName($className)
+    /**
+     * Create a ReflectionClass by name, using default reflectors etc.
+     *
+     * @param string $classNameOrInstance
+     * @return ReflectionClass
+     */
+    public static function createFromName($classNameOrInstance)
     {
-        return ClassReflector::buildDefaultReflector()->reflect($className);
+        if (gettype($classNameOrInstance) === 'object') {
```

Comment:
`is_object()`. I also suggest not having this method support object AND class, and instead having a different endpoint for that. This will be useful when, switching to PHP7, a `string` hint is added